external-crates/move/crates/move-compiler/src/naming/translate.rs

```diff
@@ -1306,10 +1347,36 @@ fn type_(context: &mut Context, sp!(loc, ety_): E::Type) -> N::Type {
                 assert!(context.env.has_errors());
                 NT::UnresolvedError
             }
+            RT::Hole => match case {
+                TypeAnnotation::StructField
+                | TypeAnnotation::ConstantSignature
+                | TypeAnnotation::FunctionSignature => {
+                    let case_str = match case {
+                        TypeAnnotation::StructField => "Struct fields",
+                        TypeAnnotation::ConstantSignature => "Constants",
+                        TypeAnnotation::FunctionSignature => "Functions",
+                        _ => unreachable!(),
+                    };
+                    let msg = format!(
+                        "Invalid usage of a placeholder for type inference '_'. \
+                        {case_str} require fully specified types. Replace '_' with a specific type"
```

Comment:
I'm not sure what this means. You cannot annotate a type variable?

Reply:
This is more that in something like
```
public struct X { f: _ }
```

You'll get this error, and I was just wondering if we should also mention in the error that instead of setting a concrete type they can use a type variable instead too:
```
public struct X<T> { f: T }
```

Reply:
Bah... yeah its probably a good suggestion, though we don't really have the ability to suggest these fixes succinctly at the moment. I'm just trying to think how to suggest adding a new type parameter instead... I'll think about it